include/ygm/container/reducing_adapter.hpp

```diff
@@ -0,0 +1,41 @@
+// Copyright 2019-2021 Lawrence Livermore National Security, LLC and other YGM
+// Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: MIT
+
+#pragma once
+#include <ygm/container/detail/container_traits.hpp>
+#include <ygm/detail/ygm_traits.hpp>
+
+namespace ygm::container {
+
+template <typename Container, typename ReductionOp>
+class reducing_adapter {
+ public:
+  reducing_adapter(Container &c, ReductionOp reducer) : m_container(c) {}
+
+  void async_reduce(const typename Container::key_type   &key,
+                    const typename Container::value_type &value) {
+    ReductionOp *r;
```

Comment:
I'd prefer if you actually stored the ReductionOp as a member and used it, rather than this pointer here.

Reply:
I concur